### PR TITLE
layers: Fix assumption on DRM modifiers

### PR DESCRIPTION
**Pull request by djdeath**
There is a confusion that DRM modifiers are bit fields but they're more like enums.

### CODE REVIEW

**Review by sfricke-samsung — APPROVED** [review]
I wrote `VK_EXT_image_drm_format_modifier` blind (didn't have anything to run it on) so feel free to make any changes needed (these all look fine to me)

File: layers/buffer_validation.cpp [hunk 1]
```diff
@@ -1629,7 +1629,7 @@ bool CoreChecks::ValidateImageFormatFeatures(const VkImageCreateInfo *pCreateInf
     bool skip = false;
 
     // validates based on imageCreateFormatFeatures from vkspec.html#resources-image-creation-limits
-    VkFormatFeatureFlags tiling_features = VK_FORMAT_FEATURE_FLAG_BITS_MAX_ENUM;
+    VkFormatFeatureFlags tiling_features = 0;
```
Comment (sfricke-samsung) [inline layers/buffer_validation.cpp:1632]: so I wish I would have wrote a comment WHY I used `VK_FORMAT_FEATURE_FLAG_BITS_MAX_ENUM` but I use it in a few other places for the same issue... curious if worth changing the other use cases of `VK_FORMAT_FEATURE_FLAG_BITS_MAX_ENUM` to `0` as well

Comment (sfricke-samsung) [reply layers/buffer_validation.cpp:1632]: looking more at it, I think I was worried my logic for `VK_IMAGE_TILING_DRM_FORMAT_MODIFIER_EXT` would not work and throw a false error for `VUID-VkImageViewCreateInfo-None-02273`

Comment (djdeath) [reply layers/buffer_validation.cpp:1632]: `0` sounds like the right value to have if the implementation reports no feature or doesn't list the drm modifier. I'll update the other location.

Comment (ncesario-lunarg) [reply layers/buffer_validation.cpp:1632]: I like the idea of using a known constant such as `VK_FORMAT_FEATURE_FLAG_BITS_MAX_ENUM` as it's not clear (to me, at least) what `0` represents (e.g., is it a valid enum value?). If `0` makes sense to everyone though, I'm fine with it.

File: layers/buffer_validation.cpp [hunk 2]
```diff
@@ -1645,35 +1645,34 @@ bool CoreChecks::ValidateImageFormatFeatures(const VkImageCreateInfo *pCreateInf
         }
 #endif
     } else if (image_tiling == VK_IMAGE_TILING_DRM_FORMAT_MODIFIER_EXT) {
-        uint64_t drm_format_modifier = 0;
+        std::set<uint64_t> drm_format_modifiers;
```
Comment (ncesario-lunarg) [inline layers/buffer_validation.cpp:1648]: If possible, this should probably be a `layer_data::unordered_set`.

**Review by ncesario-lunarg — CHANGES_REQUESTED** [review]
Just my 2 cents on using 0 as an invalid value and a type change request, but otherwise LGTM!

File: layers/buffer_validation.cpp (hunk 1 above)
Comment (sfricke-samsung) [reply layers/buffer_validation.cpp:1632]: So there is a 3 way if/else case and `tiling_features` should have been set by the end of it. I guess while I agree `VK_FORMAT_FEATURE_FLAG_BITS_MAX_ENUM` shows you it is invalid, we were never checking/asserting it anyway which is why I was ok with `0`

File: layers/buffer_validation.cpp (hunk 2 above)
Comment (djdeath) [reply layers/buffer_validation.cpp]: Done.

**Comment by djdeath** [comment]
Contribution from Intel Corporation (still tried to sign the CLA but it's not working...)

File: layers/buffer_validation.cpp [hunk 3]
```diff
@@ -1645,35 +1646,34 @@ bool CoreChecks::ValidateImageFormatFeatures(const VkImageCreateInfo *pCreateInf
         }
 #endif
     } else if (image_tiling == VK_IMAGE_TILING_DRM_FORMAT_MODIFIER_EXT) {
-        uint64_t drm_format_modifier = 0;
+        std::unordered_set<uint64_t> drm_format_modifiers;
```
Comment (ncesario-lunarg) [inline layers/buffer_validation.cpp:1649]: Did the `layer_data::unordered_set` version not work?

Comment (djdeath) [reply layers/buffer_validation.cpp:1649]: Ah sorry, I got them mixed :)

**Comment by ncesario-lunarg** [comment]
I think all that's missing is CLA signature, and then this is GTG :)

**Comment by djdeath** [comment]
> I think all that's missing is CLA signature, and then this is GTG :)

Like I mentioned above, this is a contribution from Intel Corporation. I have tried the CLA signature many times, it doesn't work.
Could it be that it's because I'm using my personal github account which is not linked to my email address?

I have trouble opening PRs with my other account because it's used for internal Intel things :(

**Comment by Khronoswebmaster** [comment]
Hi @djdeath, it appears the commit was done from account @llandwerlin-intel, and that account will need to sign the CLA.

**Comment by llandwerlin-intel** [comment]
> Hi @djdeath, it appears the commit was done from account @llandwerlin-intel, and that account will need to sign the CLA.

Makes sense now, done.

**Review by ncesario-lunarg — APPROVED** [review]

**Pull request merged**